src/lib/parsers/htmlHelper.ts

```diff
@@ -36,11 +36,100 @@ function generateTable(receivers: Receiver[], section: string = null) : string {
 	return htmlTable;
 }
 
+function generateCategoriesTable(receivers: Result[], section: string): string {
+	let htmlTable =
+		`					<table id="section-${section}" class="table pb-3 collapse">`;
+	receivers.forEach((result: Result, index: number) => {
+		const subsectionId = `section-${section}-${index}`;
+		htmlTable += `
+						<thead class="thead-light clickable" data-toggle="collapse" data-target="#${subsectionId}" area-expanded="true" aria-controls="${subsectionId}">
+							<tr><th colspan="3"><i class="fas fa-plus"></i><i class="fas fa-minus"></i> ${result.personName} &mdash; ${result.validPegsReceived.length}</th></tr>
+						</thead>
+						<tbody id="${subsectionId}" class="collapse show">`;
+		result.validPegsReceived.sort((a, b) => a.senderName.localeCompare(b.senderName));
+		result.validPegsReceived.forEach((peg: Peg) => {
+			htmlTable += `
+							<tr><td>${peg.senderName}</td><td>${peg.comment}</td><td>${peg.categories.join(', ')}</td></tr>
+`;
+		});
+
+		htmlTable +=
+			`						</tbody>
+`;
+	});
+
+	htmlTable +=
+		`					</table>`;
+
+	return htmlTable;
+}
+
+function generatePenaltyTable(receivers: Result[]): string {
+	let htmlTable =
+		`					<table id="section-penalties" class="table pb-3 collapse">`;
+
+	receivers.forEach((result: Result, index: number) => {
+		if (result.penaltyPegsGiven.length !== 0) {
+			const subsectionId = `section-penalties-${index}`;
+
+			htmlTable += `
+
+						<thead class="thead-light clickable" data-toggle="collapse" data-target="#${subsectionId}" aria-expanded="true" aria-controls="${subsectionId}">
+							<tr><th colspan="3"><i class="fas fa-plus"></i><i class="fas fa-minus"></i> ${result.personName ? result.personName : 'somebody shameful'} &mdash; ${result.penaltyPegsGiven.length}</th></tr>
+						</thead>
+						<tbody id="${subsectionId}" class="collapse show">`;
+
+			result.penaltyPegsGiven.sort((a, b) => a.receiverName.localeCompare(b.receiverName));
```

Comment:
As this one sort of relies on penalty pegs given I sort of understand why it is different, but at the same time I feel like we could translate all of these into an intermediary format so that they could use the same html helper function. That would make this far easier to maintain if we need to make changes to the html.